hack/verify-golangci-lint.sh

```diff
@@ -18,7 +18,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-VERSION=v1.23.8
+VERSION=v1.31.0
```

Comment:
This should be captured in `dependencies.yaml`.

Reply:
Added :+1: